tendermint testnet fail

<!-- Thanks for filing an issue! Before hitting the button, please answer these questions.-->

**Is this a BUG REPORT or FEATURE REQUEST?** (choose one):
bug
<!--
If this is a BUG REPORT, please:
  - Fill in as much of the template below as you can.

If this is a FEATURE REQUEST, please:
  - Describe *in detail* the feature/behavior/change you'd like to see.

In both cases, be ready for followup questions, and please respond in a timely
manner. We might ask you to provide additional logs and data (tendermint & app)
in a case of bug.
-->

**Tendermint version**: 0.16.0
**ABCI app** :0.9.0
**Environment**:
- **OS** macOs13.0.0
- **Install tools**:glide
- **Others**:


**What happened**:
tendermint testnet
panic: Panicked on a Crisis: open mytestnet/mach0/config/080033051: no such file or directory

**What you expected to happen**:
4 nodes

**How to reproduce it** (as minimally and precisely as possible):


**Anything else do we need to know**:
other commands is all right（tendetmint node，tendermint init and so on）

--- a/cmd/tendermint/commands/testnet.go
+++ b/cmd/tendermint/commands/testnet.go
@@ -75,15 +75,16 @@ func testnetFiles(cmd *cobra.Command, args []string) {
 
 // Initialize per-machine core directory
 func initMachCoreDirectory(base, mach string) error {
+	// Create priv_validator.json file if not present
+	defaultConfig := cfg.DefaultBaseConfig()
 	dir := filepath.Join(base, mach)
-	err := cmn.EnsureDir(dir, 0777)
+	privValPath := filepath.Join(dir, defaultConfig.PrivValidator)
+	dir = filepath.Dir(privValPath)
+	err := cmn.EnsureDir(dir, 0700)
 	if err != nil {
 		return err
 	}
-
-	// Create priv_validator.json file if not present
-	defaultConfig := cfg.DefaultBaseConfig()
-	ensurePrivValidator(filepath.Join(dir, defaultConfig.PrivValidator))
+	ensurePrivValidator(privValPath)
 	return nil
 
 }

--- a/types/priv_validator.go
+++ b/types/priv_validator.go
@@ -98,6 +98,7 @@ type PrivValidator interface {
 // PrivValidatorFS implements PrivValidator using data persisted to disk
 // to prevent double signing. The Signer itself can be mutated to use
 // something besides the default, for instance a hardware signer.
+// NOTE: the directory containing the privVal.filePath must already exist.
 type PrivValidatorFS struct {
 	Address       Address          `json:"address"`
 	PubKey        crypto.PubKey    `json:"pub_key"`
@@ -217,18 +218,17 @@ func (privVal *PrivValidatorFS) Save() {
 }
 
 func (privVal *PrivValidatorFS) save() {
-	if privVal.filePath == "" {
-		cmn.PanicSanity("Cannot save PrivValidator: filePath not set")
+	outFile := privVal.filePath
+	if outFile == "" {
+		panic("Cannot save PrivValidator: filePath not set")
 	}
 	jsonBytes, err := json.Marshal(privVal)
 	if err != nil {
-		// `@; BOOM!!!
-		cmn.PanicCrisis(err)
+		panic(err)
 	}
-	err = cmn.WriteFileAtomic(privVal.filePath, jsonBytes, 0600)
+	err = cmn.WriteFileAtomic(outFile, jsonBytes, 0600)
 	if err != nil {
-		// `@; BOOM!!!
-		cmn.PanicCrisis(err)
+		panic(err)
 	}
 }
 

